pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -650,3 +761,43 @@ func osdOnSDNFlag(hostnetwork bool, v cephver.CephVersion) []string {
 
 	return args
 }
+
+func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, pvcIndex, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
+	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, pvcIndex, setIndex)
+
+	// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
+	pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
+
+	// Add pvcTemplate name to pvc name. i.e <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>-<pvcTemplateName>
+	if len(pvcTemplate.GetName()) != 0 {
+		pvcGenerateName = pvcGenerateName + pvcTemplate.GetName() + "-"
```

Comment:
Do we need the template name to be included in the pvc name? Right now we're only going to use the first template and in the future we're going to have specific names we look for to implement the advanced scenarios with multiple devices per OSD. So we can skip the template name for now.

Reply:
cc @ashishranjan738

Reply:
ok

Reply:
I still think we can keep it as it will help us identify the pvc's template easily.